### PR TITLE
Improve documentation for X-Opaque-ID

### LINKED ISSUE #44851: Improve documentation on X-Opaque-ID
I couldn't find in docs that X-Opaque-ID can be used to identify origin of slow queries or deprecation requests. This should be improved

### PR DESCRIPTION
**Pull request by pgomulka**
this field can be present in search slow logs and deprecation logs. The
docs describes how to enable this functionality and what expect in logs.
closes #44851

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

**Comment by pgomulka** [comment]
ok to test

**Comment by pgomulka** [comment]
@jrodewig would you be able to have a look? I am having problems to passing a CI for this. Is there an issue with docs build atm?

File: docs/reference/index-modules/slowlog.asciidoc [hunk 1]
```diff
@@ -83,6 +83,36 @@ logger.index_search_slowlog_rolling.appenderRef.index_search_slowlog_rolling.ref
 logger.index_search_slowlog_rolling.additivity = false
 --------------------------------------------------
 
+==== Identifying search slow log origin
+
+It is often useful to identify what triggered a slow running query. If a call was initiated with an `X-Opaque-ID` header, then this value
+will be present in Search Slow logs as an additional **id** field (scroll to the right).
+```
```
Comment (jrodewig) [inline docs/reference/index-modules/slowlog.asciidoc]: I'd rewrite this to use the Asciidoc source code blocks.

For example:
```
[source,txt]
----
[2019-08-30T11:59:37,786][WARN ][i.s.s.query              ] [node-0] [index6][0] took[78.4micros], took_millis[0], total_hits[0 hits], stats[], search_type[QUERY_THEN_FETCH], total_shards[1], source[{"query":{"match_all":{"boost":1.0}}}], id[MY_USER_ID],
----
```

Also consider replacing `[2019-08-30T11:59:37,786]` with a more future-proof timestamp like `[2030-08-30T11:59:37,786]`

File: docs/reference/index-modules/slowlog.asciidoc [hunk 2]
```diff
@@ -83,6 +83,36 @@ logger.index_search_slowlog_rolling.appenderRef.index_search_slowlog_rolling.ref
...
+[2019-08-30T11:59:37,786][WARN ][i.s.s.query              ] [node-0] [index6][0] took[78.4micros], took_millis[0], total_hits[0 hits], stats[], search_type[QUERY_THEN_FETCH], total_shards[1], source[{"query":{"match_all":{"boost":1.0}}}], id[MY_USER_ID],
+```
+the same applies to JSON logs
+```
```
Comment (jrodewig) [inline docs/reference/index-modules/slowlog.asciidoc]: Also replace with a code block.
```
[source,js]
----
{
  "type": "index_search_slowlog",
  "timestamp": "2019-08-30T11:59:37,786+02:00",
  "level": "WARN",
  "component": "i.s.s.query",
  "cluster.name": "distribution_run",
  "node.name": "node-0",
  "message": "[index6][0]",
  "took": "78.4micros",
  "took_millis": "0",
  "total_hits": "0 hits",
  "stats": "[]",
  "search_type": "QUERY_THEN_FETCH",
  "total_shards": "1",
  "source": "{\"query\":{\"match_all\":{\"boost\":1.0}}}",
  "id": "MY_USER_ID",
  "cluster.uuid": "Aq-c-PAeQiK3tfBYtig9Bw",
  "node.id": "D7fUYfnfTLa2D7y-xw6tZg"
}
----
// NOTCONSOLE
```

File: docs/reference/setup/logging-config.asciidoc [hunk 3]
```diff
@@ -214,6 +214,22 @@ files (four rolled logs, and the active log).
 You can disable it in the `config/log4j2.properties` file by setting the deprecation
 log level to `error`.
 
+You can identify who is triggering deprecated functionality if `X-Opaque-Id` was used as an HTTP header.
+This field will be then present as `X-Opaque-ID` field in deprecation JSON logs.
+```
```
Comment (jrodewig) [inline docs/reference/setup/logging-config.asciidoc]: Replace with a code block

```
[source,js]
----
{
  "type": "deprecation",
  "timestamp": "2030-08-30T12:07:07,126+02:00",
  "level": "WARN",
  "component": "o.e.d.r.a.a.i.RestCreateIndexAction",
  "cluster.name": "distribution_run",
  "node.name": "node-0",
  "message": "[types removal] Using include_type_name in create index requests is deprecated. The parameter will be removed in the next major version.",
  "x-opaque-id": "MY_USER_ID",
  "cluster.uuid": "Aq-c-PAeQiK3tfBYtig9Bw",
  "node.id": "D7fUYfnfTLa2D7y-xw6tZg"
}
----
````

File: docs/reference/index-modules/slowlog.asciidoc [hunk 4]
```diff
@@ -83,6 +83,36 @@ logger.index_search_slowlog_rolling.appenderRef.index_search_slowlog_rolling.ref
 logger.index_search_slowlog_rolling.additivity = false
 --------------------------------------------------
 
+==== Identifying search slow log origin
+
+It is often useful to identify what triggered a slow running query. If a call was initiated with an `X-Opaque-ID` header, then this value
+will be present in Search Slow logs as an additional **id** field (scroll to the right).
```
Comment (jrodewig) [inline docs/reference/index-modules/slowlog.asciidoc]:
```suggestion
is included in Search Slow logs as an additional **id** field (scroll to the right).
```

File: docs/reference/index-modules/slowlog.asciidoc [hunk 5]
```diff
@@ -83,6 +83,36 @@ logger.index_search_slowlog_rolling.appenderRef.index_search_slowlog_rolling.ref
 logger.index_search_slowlog_rolling.additivity = false
 --------------------------------------------------
 
+==== Identifying search slow log origin
+
+It is often useful to identify what triggered a slow running query. If a call was initiated with an `X-Opaque-ID` header, then this value
```
Comment (jrodewig) [inline docs/reference/index-modules/slowlog.asciidoc]:
```suggestion
It is often useful to identify what triggered a slow running query. If a call was initiated with an `X-Opaque-ID` header, then the user ID
```

File: docs/reference/index-modules/slowlog.asciidoc [hunk 6]
```diff
@@ -83,6 +83,36 @@ logger.index_search_slowlog_rolling.appenderRef.index_search_slowlog_rolling.ref
...
+will be present in Search Slow logs as an additional **id** field (scroll to the right).
+```
+[2019-08-30T11:59:37,786][WARN ][i.s.s.query              ] [node-0] [index6][0] took[78.4micros], took_millis[0], total_hits[0 hits], stats[], search_type[QUERY_THEN_FETCH], total_shards[1], source[{"query":{"match_all":{"boost":1.0}}}], id[MY_USER_ID],
+```
+the same applies to JSON logs
```
Comment (jrodewig) [inline docs/reference/index-modules/slowlog.asciidoc]:
```suggestion
The user ID is also included in JSON logs.
```

File: docs/reference/setup/logging-config.asciidoc [hunk 7]
```diff
@@ -214,6 +214,22 @@ files (four rolled logs, and the active log).
 You can disable it in the `config/log4j2.properties` file by setting the deprecation
 log level to `error`.
 
+You can identify who is triggering deprecated functionality if `X-Opaque-Id` was used as an HTTP header.
```
Comment (jrodewig) [inline docs/reference/setup/logging-config.asciidoc]:
```suggestion
You can identify what is triggering deprecated functionality if `X-Opaque-Id` was used as an HTTP header.
```

File: docs/reference/setup/logging-config.asciidoc [hunk 8]
```diff
@@ -214,6 +214,22 @@ files (four rolled logs, and the active log).
...
+This field will be then present as `X-Opaque-ID` field in deprecation JSON logs.
```
Comment (jrodewig) [inline docs/reference/setup/logging-config.asciidoc]:
```suggestion
The user ID is included in the `X-Opaque-ID` field in deprecation JSON logs.
```

**Comment by jrodewig** [comment]
@elaticmachine run elasticsearch-ci/docs

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Review by jrodewig — CHANGES_REQUESTED** [review]
Left a few suggestions about changing the snippets to Asciidoc code blocks.

Also left some less important editorial changes. Please feel free to take/discard those as you see fit. 

Thanks!

File: docs/reference/index-modules/slowlog.asciidoc [hunk 9]
```diff
@@ -83,6 +83,36 @@ logger.index_search_slowlog_rolling.appenderRef.index_search_slowlog_rolling.ref
 logger.index_search_slowlog_rolling.additivity = false
 --------------------------------------------------
 
+==== Identifying search slow log origin
```
Comment (jrodewig) [inline docs/reference/index-modules/slowlog.asciidoc]:
```suggestion
[float]
==== Identifying search slow log origin
```

**Review by jrodewig — APPROVED** [review]
LGTM. Thanks @pgomulka!

**Comment by luyuncheng** [comment]
LGTM

**Pull request merged**